core/iwasm/libraries/lib-socket/test/nslookup.c

```diff
@@ -0,0 +1,16 @@
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#else
+#include <netdb.h>
+#endif
+
+int
+main()
+{
+    struct addrinfo *result;
+    int ret = getaddrinfo("https://google.com", NULL, NULL, &result);
+    assert(ret >= 0);
```

Comment:
can `getaddrinfo` ever return value greater than zero? And if so, is that successful error code? According to the [doc](https://man7.org/linux/man-pages/man3/getaddrinfo.3.html#RETURN_VALUE):

> getaddrinfo() returns 0 if it succeeds, or one of the following nonzero error codes:

So we probably want to change the assertion to:
```suggestion
    assert(ret == 0);
```